InkToolbar Style updates

viveksharmadev opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
InkToolbar style updates

## Motivation and Context
This change is needed to make inkToolbar to look like other winui controls

## How Has This Been Tested?
Verified the changes visually 

## Screenshots (if appropriate):
![image](https://user-images.githubusercontent.com/16787170/113059506-61e3ac80-9164-11eb-920a-1ce527114a87.png)

![image](https://user-images.githubusercontent.com/16787170/113059548-70ca5f00-9164-11eb-9eb2-fd05f6cee69e.png)

![image](https://user-images.githubusercontent.com/16787170/113059568-758f1300-9164-11eb-9495-1fe29fe35544.png)

![image](https://user-images.githubusercontent.com/16787170/113059589-79bb3080-9164-11eb-8585-308092243bcd.png)

![image](https://user-images.githubusercontent.com/16787170/113059614-82ac0200-9164-11eb-997a-e95e74972d33.png)

![image](https://user-images.githubusercontent.com/16787170/113059633-86d81f80-9164-11eb-96ac-01d569e88954.png)

![image](https://user-images.githubusercontent.com/16787170/113059651-8dff2d80-9164-11eb-8521-f543385492e7.png)

![image](https://user-images.githubusercontent.com/16787170/113059678-97889580-9164-11eb-907a-21d1e61f3676.png)

![image](https://user-images.githubusercontent.com/16787170/113059690-9bb4b300-9164-11eb-9811-5b769193cc71.png)

![image](https://user-images.githubusercontent.com/16787170/113059708-a0796700-9164-11eb-8a1e-94e18225eb71.png)

![image](https://user-images.githubusercontent.com/16787170/113059737-a66f4800-9164-11eb-85a2-180309fdaeb2.png)

![image](https://user-images.githubusercontent.com/16787170/113060266-42994f00-9165-11eb-9181-ff1a2f7801f8.png)

![image](https://user-images.githubusercontent.com/16787170/113059819-bd159f00-9164-11eb-944b-2744ba69bab8.png)

ranjeshj commented:
/azp run

teaP commented on dev/CommonStyles/InkToolbar_themeresources.xaml:
```diff
@@ -299,7 +301,7 @@
                                 <VisualState x:Name="ShowExtensionGlyph">
                                     <VisualState.Setters>
-                                        <Setter Target="ExtensionGlyph.Opacity" Value="1" />
+                                        <Setter Target="ExtensionGlyph.Opacity" Value="0" />
```
Is this glyph never supposed to be visible, then?

viveksharmadev replied:
Now we want to make it visible, earlier the ask was not to be visible. Will update this accordingly!

teaP reviewed (changes requested):
Can you please also include some screenshots?

ranjeshj commented:
Looks like we need platform checks for BackgroundTransition and BrushTransition

viveksharmadev commented:
> Can you please also include some screenshots?

FireCubeStudios commented:
I would like to see a screen shot of the toolbar and flyout. it would be usefull since i have inking apps and would like to know hwo the toolbar would look like

ranjeshj commented:
>             <StaticResource x:Key="InkToolbarFlyoutItemBorderPressedThemeBrush" ResourceKey="InkToolbarFlyoutItemBorderSelectedThemeBrush" />            

nit:stray spaces

---
Refers to: dev/CommonStyles/InkToolbar_themeresources.xaml:52 in ded9926. [](commit_id = ded99262844b9af223cf25661137f8a9eb061674, deletion_comment = False)

ranjeshj commented:
>                         CornerRadius="{TemplateBinding CornerRadius}" >

follow the same pattern as everywhere with Contract7Present/NotPresent for CornerRadius

---
Refers to: dev/CommonStyles/InkToolbar_themeresources.xaml:284 in ded9926. [](commit_id = ded99262844b9af223cf25661137f8a9eb061674, deletion_comment = False)

ranjeshj commented:
>                                 CornerRadius="{TemplateBinding CornerRadius}"/>

Contract7Present/NotPresent needs to be added for CornerRadius.

---
Refers to: dev/CommonStyles/InkToolbar_themeresources.xaml:504 in ded9926. [](commit_id = ded99262844b9af223cf25661137f8a9eb061674, deletion_comment = False)

ranjeshj commented:
>                         CornerRadius="{TemplateBinding CornerRadius}">

Contract7Present/NotPresent needs to be added for CornerRadius. Control.CornerRadius was added in contract 7. Grid/Border had it from before so dont need for those.

---
Refers to: dev/CommonStyles/InkToolbar_themeresources.xaml:783 in ded9926. [](commit_id = ded99262844b9af223cf25661137f8a9eb061674, deletion_comment = False)

ranjeshj commented on dev/CommonStyles/TestUI/InkToolbarPage.xaml:
```diff
@@ -5,7 +5,7 @@
     Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">
     <StackPanel>
-        <InkToolbar TargetInkCanvas="{x:Bind inkCanvas}"/>
+        <InkToolbar TargetInkCanvas="{x:Bind inkCanvas}" Margin="2,0,0,0"/>
```
We should add a test to load and show this page as well as click the flyouts to open them, so we ensure this runs in the lab in all OS versions.

ranjeshj replied:
Should this margin be on the InkToolbar itself ?

ranjeshj commented:
>                         CornerRadius="{TemplateBinding CornerRadius}">

Contract7Present/NotPresent needs to be added for CornerRadius.

---
Refers to: dev/CommonStyles/InkToolbar_themeresources.xaml:534 in ded9926. [](commit_id = ded99262844b9af223cf25661137f8a9eb061674, deletion_comment = False)

pengzh0928 commented on dev/CommonStyles/InkToolbar_themeresources.xaml:
```diff
@@ -131,20 +132,39 @@
+    <x:Double x:Key="InkToolbarButtonWidth">40</x:Double>
+    <x:Double x:Key="InkToolbarButtonHeight">40</x:Double>
+    <x:Double x:Key="InkToolbarButtonContentSize">16</x:Double>
+    <Thickness x:Key="InkToolbarFlyoutPanelMargin">0,1,0,2</Thickness>
```
Don't forget update as we discuss

pengzh0928 commented on dev/CommonStyles/InkToolbar_themeresources.xaml:
```diff
@@ -411,7 +500,8 @@
+                                CornerRadius="{TemplateBinding CornerRadius}"/>
 
                         </Grid>
                         <Rectangle x:Name="SelectionAccent" Style="{StaticResource InkToolbarSelectionAccentStyle}" />
```
I think currently, we don't need ExtensionGlyph any more. Is that correct? If yes, we should remove all of them in the code.

pengzh0928 commented on dev/CommonStyles/InkToolbar_themeresources.xaml:
```diff
@@ -1279,7 +1580,7 @@
                                 <VisualState x:Name="ShowExtensionGlyph">
                                     <VisualState.Setters>
-                                        <Setter Target="ExtensionGlyph.Opacity" Value="1" />
+                                        <Setter Target="ExtensionGlyph.Opacity" Value="0" />
```
Same here, this behavior is too tricky. We should remove all ExtensionGlyph related code.

pengzh0928 commented on dev/CommonStyles/InkToolbar_themeresources.xaml:
```diff
@@ -1937,7 +2318,8 @@
                                 </VisualState>
-                                <VisualState x:Name="HideExtensionGlyph" />
+                                <!--Without commenting this one, glyph is getting invisible when pressing the button-->
+                                <!--<VisualState x:Name="HideExtensionGlyph" />-->
```
You also could add another way here. Make the HideExtensionGlyph as same as ShowExtensionGlyph

pengzh0928 replied:
Currently, we don't need ExtensionGlyph any more. Why ExtensionGlyph will influence SelectionAccent?

pengzh0928 commented on dev/CommonStyles/InkToolbar_themeresources.xaml:
```diff
@@ -2038,7 +2420,8 @@
+                                CornerRadius="{TemplateBinding CornerRadius}"/>
 
                         </Grid>
                         <Rectangle x:Name="SelectionAccent" Style="{StaticResource InkToolbarSelectionAccentStyle}" />
```
Doe SelectionAccent already support vertical origin?

pengzh0928 reviewed (changes requested):
Now my concern have two parts: 1. vertical support. 2. From the screenshot, we just need SelectionAccent, but we still keep a mount of ExtensionGlyph in the code. I believe we should remove it.

viveksharmadev commented:
> I would like to see a screen shot of the toolbar and flyout. it would be usefull since i have inking apps and would like to know hwo the toolbar would look like



> > ```
> >                     CornerRadius="{TemplateBinding CornerRadius}">
> > ```
> 
> Contract7Present/NotPresent needs to be added for CornerRadius.
> 
> Refers to: dev/CommonStyles/InkToolbar_themeresources.xaml:534 in [ded9926](https://github.com/microsoft/microsoft-ui-xaml/commit/ded99262844b9af223cf25661137f8a9eb061674). [](commit_id = [ded9926](https://github.com/microsoft/microsoft-ui-xaml/commit/ded99262844b9af223cf25661137f8a9eb061674), deletion_comment = False)

Addressed the above issues.

viveksharmadev commented:
> > ```
> >                     CornerRadius="{TemplateBinding CornerRadius}">
> > ```
> 
> Contract7Present/NotPresent needs to be added for CornerRadius.
> 
> Refers to: dev/CommonStyles/InkToolbar_themeresources.xaml:534 in [ded9926](https://github.com/microsoft/microsoft-ui-xaml/commit/ded99262844b9af223cf25661137f8a9eb061674). [](commit_id = [ded9926](https://github.com/microsoft/microsoft-ui-xaml/commit/ded99262844b9af223cf25661137f8a9eb061674), deletion_comment = False)

Addressed the above issues

viveksharmadev commented:
> > ```
> >         <StaticResource x:Key="InkToolbarFlyoutItemBorderPressedThemeBrush" ResourceKey="InkToolbarFlyoutItemBorderSelectedThemeBrush" />            
> > ```
> 
> nit:stray spaces
> 
> Refers to: dev/CommonStyles/InkToolbar_themeresources.xaml:52 in [ded9926](https://github.com/microsoft/microsoft-ui-xaml/commit/ded99262844b9af223cf25661137f8a9eb061674). [](commit_id = [ded9926](https://github.com/microsoft/microsoft-ui-xaml/commit/ded99262844b9af223cf25661137f8a9eb061674), deletion_comment = False)

Addressed the above issue

pengzh0928 commented on dev/CommonStyles/InkToolbar_themeresources.xaml:
```diff
@@ -1,114 +1,115 @@
+            <StaticResource x:Key="InkToolbarFlyoutItemBorderHoverThemeBrush" ResourceKey="SubtleFillColorSecondaryBrush" />
+            <SolidColorBrush x:Key="InkToolbarFlyoutItemBorderSelectedThemeBrush" Color="#FFFFFF" />
+            <StaticResource x:Key="InkToolbarFlyoutItemBorderPressedThemeBrush" ResourceKey="InkToolbarFlyoutItemBorderSelectedThemeBrush" />            
+            <StaticResource x:Key="InkToolbarAccentColorThemeBrush" ResourceKey="AccentTextFillColorPrimaryBrush" />
```
If you could move them to under <!-- Forground-->. It will be better.

---
In reply to: [599927196](https://github.com/microsoft/microsoft-ui-xaml/pull/4507#discussion_r599927196) [](ancestors = 599927196)

teaP commented on dev/CommonStyles/InkToolbar_themeresources.xaml:
```diff
@@ -214,11 +238,29 @@
+        <Setter Property="FontWeight" Value="SemiBold" />
+    </Style>
+    
     <Style x:Key="InkToolbarSelectionAccentStyle" TargetType="Rectangle">
```
If this is no longer being used, the style should be removed from this file, and all SelectionAccent rectangles should be removed from the markup since they were never be visible.

viveksharmadev replied:
Here SelectionAccent is a blue pill that was visible earlier, it was added and removed couple of times now, so put the related code, now decision of removing it seems to be final so will get rid of this!

teaP commented on dev/CommonStyles/InkToolbar_themeresources.xaml:
```diff
@@ -214,11 +238,29 @@
+        <Setter Property="Height" Value="16" />
+        <Setter Property="Width" Value="3"/>
+        <Setter Property="Fill" Value="{ThemeResource InkToolbarAccentColorForegroundThemeBrush}" />
+        <contract7Present:Setter Property="RadiusX" Value="8" />
```
8 seems awfully high.... shouldn't this be 1.5 too?

viveksharmadev replied:
It is half of Height, else round corner are not getting visible correctly.

teaP replied:
Being half of height would make it an oval of some kind. That doesn't make sense.

teaP commented on dev/CommonStyles/InkToolbar_themeresources.xaml:
```diff
@@ -272,23 +313,71 @@
                                     </VisualState.Setters>
                                 </VisualState>
 
+                            </VisualStateGroup>
```
You can probably remove all this (and in the other places where they appear as well).

viveksharmadev replied:
I can not see here, are they related to selectionAccent?

teaP replied:
I mean the empty VisualStateGroups.

teaP commented on dev/CommonStyles/InkToolbar_themeresources.xaml:
```diff
@@ -1003,8 +1238,10 @@
-        <Setter Property="Height" Value="48" />
+        <Setter Property="Height" Value="40" />
         <Setter Property="UseSystemFocusVisuals" Value="{StaticResource UseSystemFocusVisuals}" />
+        <!--<Setter Property="Margin" Value = "2,4,2,2" />-->
```
please remove if this isn't going to be used.

teaP commented on dev/CommonStyles/InkToolbar_themeresources.xaml:
```diff
@@ -1016,27 +1253,31 @@
                                     <VisualState.Setters>
                                         <Setter Target="Border.Background" Value="{ThemeResource InkToolbarFlyoutItemPressedBackgroundThemeBrush}" />
                                         <Setter Target="Content.Foreground" Value="{ThemeResource InkToolbarFlyoutItemPressedForegroundThemeBrush}" />
+                                        <Setter Target="CurrentSelectionIcon.Opacity" Value="1" />
```
I'm not 100% sure what's going on here, but you can't set an property in two different VisualStateGroups. Should these be in the same group then?

pengzh0928 replied:
Hmmm. This is a new knowledge for me. I have no idea about "can't set an property in two different VisualStateGroups".  Let me create an email to make sure we are on the same page.

---
In reply to: [600893246](https://github.com/microsoft/microsoft-ui-xaml/pull/4507#discussion_r600893246) [](ancestors = 600893246)

pengzh0928 commented:
>                                         <Setter Target="ExtensionGlyph.Opacity" Value="0" />

Please add some comments here.

---
Refers to: dev/CommonStyles/InkToolbar_themeresources.xaml:1054 in d4a633d. [](commit_id = d4a633d74f3c01ddd8b31b05512ad992a9935c16, deletion_comment = False)

pengzh0928 commented:
>                                         <Setter Target="CurrentSelectionIcon.Opacity" Value="1" />

THis should be 0, please help me change it.

---
Refers to: dev/CommonStyles/InkToolbar_themeresources.xaml:1181 in d4a633d. [](commit_id = d4a633d74f3c01ddd8b31b05512ad992a9935c16, deletion_comment = False)

viveksharmadev commented:
> Now my concern have two parts: 1. vertical support. 2. From the screenshot, we just need SelectionAccent, but we still keep a mount of ExtensionGlyph in the code. I believe we should remove it.

Added vertical test and removed SelectionAccent, still need ExtensionGlyph, so keeping it as ususal

pengzh0928 commented on dev/CommonStyles/InteractionTests/CommonStylesTests.cs:
```diff
@@ -157,6 +157,12 @@ public void InkToolbarTest()
+                var verticalSupport = FindElement.ByName("VerticalInkToolbar");
+                Verify.IsNotNull(verticalSupport, "Verify InkToolbar page doesn't crash for vertical support test");
+
+                DropDownButton flyoutButton = FindElement.ByName<DropDownButton>("InkToolbar");
```
In truth, you haven't open flyout menu here.

pengzh0928 replied:
Why use DropDownButton here? I believe you could use var here, or use Inktoolbar sepcific control.

teaP commented on dev/CommonStyles/InkToolbar_themeresources.xaml:
```diff
@@ -1217,32 +1321,31 @@
-                        Height="{ThemeResource InkToolbarButtonHeight}">
+                        Width="{StaticResource InkToolbarButtonWidth}"
+                        Height="{StaticResource InkToolbarButtonHeight}"
+                        CornerRadius="{TemplateBinding CornerRadius}">
```
Missing contrast7 here and in some other places as well.

viveksharmadev commented on dev/CommonStyles/InkToolbar_themeresources.xaml:
```diff
@@ -272,34 +303,68 @@
                                 <VisualState x:Name="ShowExtensionGlyph">
                                     <VisualState.Setters>
-                                        <Setter Target="ExtensionGlyph.Opacity" Value="1" />
+                                        <Setter Target="ExtensionGlyph.Opacity" Value="0" />
```
Removed ShowExtensionGlyph but If I set ExtensionGlyph.Opacity to 0, chevron gets disappeared

teaP commented:
/azp run

ranjeshj commented:
Error log:
[Error]: * InkToolbarButtonSelectedBackgroundPointerOver
[Error]: * InkToolbarSeelctedButtonBackgroundPointerOver
[Error]: * InkToolbarButtonSelectedBackgroundPointerOver
[Error]: * InkToolbarSeelctedButtonBackgroundPointerOver
[Error]: IsTrue - Resource Keys you have added are missing in one of the theme dictionaries. This is trouble since we might end up crashing when trying to resolve the key in that Theme. [File C:\a\1\s\dev\CommonStyles\APITests\CommonStylesTests.cs Line: 70]

viveksharmadev commented:
/azp run

viveksharmadev commented:
> Error log:
> [Error]: * InkToolbarButtonSelectedBackgroundPointerOver
> [Error]: * InkToolbarSeelctedButtonBackgroundPointerOver
> [Error]: * InkToolbarButtonSelectedBackgroundPointerOver
> [Error]: * InkToolbarSeelctedButtonBackgroundPointerOver
> [Error]: IsTrue - Resource Keys you have added are missing in one of the theme dictionaries. This is trouble since we might end up crashing when trying to resolve the key in that Theme. [File C:\a\1\s\dev\CommonStyles\APITests\CommonStylesTests.cs Line: 70]

Addressed the above issue
